docs/TOML.md

```diff
@@ -0,0 +1,100 @@
+# TOML
+
+Telegraf uses TOML as the configuration language. The following outlines a few
+common questions and issues that cause questions or confusion.
+
+## Reference and Validator
+
+For all things TOML related, please consult the [TOML Spec][] and consider
+using a TOML validator. In VSCode the [Even Better TOML][] extension or use the
+[TOML Lint][] website to validate your TOML config.
+
+[TOML Spec]: https://toml.io/en/v1.0.0
+[Even Better TOML]: https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml
+[TOML Lint]: https://www.toml-lint.com/
+
+## Multiple TOML Files
+
+TOML technically does not support multiple files, this is done as a convince for
+users.
+
+Users should be aware that when Telegraf reads a user's config, if multiple
+files or directories are read in, all files are combined into a single string
+during the launch process.
+
+## Single Table vs Array of Tables
+
+Telegraf uses a single agent table (e.g. `[agent]`) to control high-level agent
+specific configurations. This section can only be defined once for all config
+files and should be in the first file read in to take effect. This cannot be
+defined per-config file.
```

Comment:
Actually not true. It can be put into multiple files, only they will be merged and duplicated options will be overridden. It also definitely does not required to be in the first file, however it can be noted that this is more convenient to do so.

Reply:
I am not going to document that it is possible when it provides undefined behavior for some of the values.